src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/EventTypeTest.java

```diff
@@ -0,0 +1,138 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.AssignApplicantSolicitorHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.CallbackHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.FinremCallbackHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.FinremCallbackRequest;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.GeneralApplicationHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.intervener.IntervenerHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Slf4j
+public class EventTypeTest {
+
+    private final List<Class> handlerClassesToIgnore = Arrays.asList(FinremCallbackHandler.class, FinremCallbackRequest.class,
+        GeneralApplicationHandler.class, CallbackHandler.class, IntervenerHandler.class, AssignApplicantSolicitorHandler.class);
+
+    @Test
+    public void givenEventHandler_whenMoreThanOneEventMatches_thenThrowError() throws ClassNotFoundException {
+        List<String> errors = new ArrayList<>();
+        var handlerClasses = getHandlerClasses();
+        for (EventType event : EventType.values()) {
+            for (CaseType caseType : CaseType.values()) {
+                for (CallbackType callbackType : CallbackType.values()) {
+                    List<Class> matchingClasses = new ArrayList<>();
+                    findMatchingClasses(event, caseType, callbackType, handlerClasses, matchingClasses);
+                    if (matchingClasses.size() > 1) {
+                        errors.add("The combination of event type " + event + ", case type " + caseType
+                            + " and callback type " + callbackType + " has been found in more than one class " + matchingClasses);
+                    }
+                }
+            }
+        }
+        errors.forEach(log::error);
+        assert errors.isEmpty();
```

Comment:
sorry, I should have been more clear, when writing test assertions, use the org.hamcrest.MatcherAssert.assertThat